pkg/proxy/iptables/proxier.go

```diff
@@ -438,18 +429,36 @@ func (proxier *Proxier) SyncLoop() {
 	}
 }
 
-// OnServiceUpdate tracks the active set of service proxies.
-// They will be synchronized using syncProxyRules()
-func (proxier *Proxier) OnServiceUpdate(allServices []api.Service) {
-	start := time.Now()
-	defer func() {
-		glog.V(4).Infof("OnServiceUpdate took %v for %d services", time.Since(start), len(allServices))
-	}()
-	proxier.mu.Lock()
-	defer proxier.mu.Unlock()
-	proxier.haveReceivedServiceUpdate = true
+// Tests whether two maps are equivalent with an initial size check optimization
+func mapsEquiv(lhs, rhs interface{}) bool {
+	map1, ok1 := lhs.(map[interface{}]interface{})
+	map2, ok2 := rhs.(map[interface{}]interface{})
+	if !ok1 || !ok2 {
+		glog.Warningf("Failed to convert map")
+		return false
+	}
 
-	activeServices := make(map[proxy.ServicePortName]bool) // use a map as a set
+	if len(map1) != len(map2) {
```

Comment:
It looks like the code in DeepEqual (once it determines it is a map in the helper function it calls) tests the length fairly early on in the function.

Is there an advantage to having this function rather than just calling DeepEqual directly?